core/common/src/test/scala/org/typelevel/otel4s/context/propagation/PassThroughPropagatorSuite.scala

```diff
@@ -36,4 +36,11 @@ class PassThroughPropagatorSuite extends FunSuite {
     val injected = propagator.inject(extracted, Map.empty[String, String])
     assert(injected.isEmpty)
   }
+
+  test("does not inject fields after extracting none") {
```

Comment:
I tried to replicate this test with oteljava PassThroughPropagator (the test fails):
```scala
test("does not inject fields after extracting none") {
  import scala.jdk.CollectionConverters._
  val getter = new TextMapGetter[Map[String, String]] {
    def keys(carrier: Map[String, String]): java.lang.Iterable[String] = carrier.keys.toList.asJava
    def get(carrier: Map[String, String], key: String): String = carrier.get(key).orNull
  }
  val setter = new TextMapSetter[scala.collection.mutable.Map[String, String]] {
    def set(carrier: scala.collection.mutable.Map[String, String], key: String, value: String): Unit =
      carrier.update(key, value)
  }
  val propagator = PassThroughPropagator.create("foo", "bar")
  val extracted1 = propagator.extract(Context.root(), Map("foo" -> "0"), getter)
  val extracted2 = propagator.extract(extracted1, Map.empty[String, String], getter)
  println(extracted1)
  println(extracted2)
  val map = scala.collection.mutable.Map.empty[String, String]
  propagator.inject(extracted2, map, setter)
  println(map)
  assert(map.isEmpty)
}
```

The output:
```
{passthroughpropagator-keyvalues=[foo, 0]}
{passthroughpropagator-keyvalues=[foo, 0]}
HashMap(foo -> 0)
```

Reply:
that feels very wrong, I don't like that at all. the issue is, if you do

```scala
  val extracted2 = propagator.extract(extracted1, Map("bar" -> "1"), getter)
```

instead, it *will* remove `"foo" -> "0"` from the context, which seems inconsistent

Reply:
My expectation from the `PassThroughPropagator` is that it would try to extract **all** fields. Let's say the propagator is `PassThroughPropagator("foo", "bar")`, and I expect both `foo` and `bar` to be extracted from the carrier.

And a few scenarios and behaviors that I expect:
1) `carrier = Map("foo" -> "1", "bar" -> "2")` - the upstream propagated **two** headers -> `List("foo" -> "1", "bar" -> "2")` should be stored in the context
2) `carrier = Map("foo" -> "1")` - the upstream propagated only **one** header -> `List("foo" -> "1")` should be stored in the context
3) `carrier = Map()` - the upstream doesn't propagate any header -> nothing should be stored in the context

____

In some sense, it would be easier if the `PassThroughtPropagator` stored values as `List[String -> Option[String]]`.
Then, the current behavior would make more sense. 

For `carrier = Map("foo" -> "1")` it would be `Lift("foo" -> Some("1"), "bar" -> None)`.

Reply:
after chatting with @rossabaker and looking at [the spec](https://opentelemetry.io/docs/specs/otel/context/api-propagators/#extract) I think I agree with you, though the spec is not worded super clearly